src/org/opendatakit/briefcase/export/ExportConfiguration.java

```diff
@@ -327,14 +354,17 @@ public boolean isValidAsCustomConf() {
     return endDate.map(mapper);
   }
 
-  public ExportConfiguration fallingBackTo(ExportConfiguration defaultConfiguration) {
+  public ExportConfiguration fallingBackTo(ExportConfiguration defaultConfiguration, FormStatus form) {
```

Comment:
Remember that an `ExportConfiguration` object always knows if it needs a private key or not. Why not trust the value of the `defaultConfiguration` arg directly?

Reply:
Yeah Exportconfiguration always know if a form needs a private key or not. But when we reach this case it means the form in question doesn't have any custom config and hence we fall back to the default export configuration to prepare a config for the form. This happens when the user clears all fields of the form custom config dialog, removing this config from memory and thus making the form not to have any custom config.
  But as you may have noticed we are just copying fields from the default configuration to create a config for this form and as you know the field formNeedsPrivatekey for default config is set to Optional.empty(). We still need to be able to know if this form needs a private key that's why a form status is passed to the fallingBackTo method. In case the form needs a private key and the default config doesn't have a pem file set, the config for this form will be invalid and the export button will be disabled.
Remember the clear button in the Custom configuration dialog causes the custom config to be removed from memory. Actually there is a hash map that keeps track of all custom configs and we check this hash map to get a custom config for a particular form if not present we fall back to the default configuration.

Reply:
OK, I understand your point. 

The thing is that before this, we were happy just by letting it fail: you could select a form that needs decryption and not set any PEM file location. Then Briefcase fails, showing the user the error.

The `fallingBackTo(ExportConfiguration conf)` method should not know anything about forms. It's just a method to merge two configurations. That's why I'm resisting to pass the form. A form should have to know how to fallback to another form without any other information.

If the problem is that, by removing confs, we're losing the information about the need for a private key, I'd suggest to change the `ExportForms.getConfiguration(FormStatus form)` method like this:

```java
public ExportConfiguration getConfiguration(FormStatus form) {
  return Optional.ofNullable(customConfigurations.get(form.getFormDefinition().getFormId()))
      .orElse(ExportConfiguration.empty(form))
      .fallingBackTo(defaultConfiguration);
}
```

I think that adding an `ExportConfiguration.empty(FormStatus form)` factory could be better because it lets us create an empty conf for a form, which we couldn't do before. Doing so, requires some other changes on other parts, like where the conf buttons are created and the Dialog factories, etc. I've prepared a branch if you want to check it out: https://github.com/ggalmazor/briefcase/commit/ad110ea3a47333767485ec9b9eb18a4197c6680c

Note that I've change the name of the arg in `ExportConfiguration.fallingBackTo()` to avoid confusion: instead of `defaultConfiguration`, now it's called `other`, implying no assumptions on the object.